### PR TITLE
Kh.expand config file generation to improve documentation

### PR DESCRIPTION
**Pull request by wyhaines**
# Overview
This PR addresses issue #684. We used the current documentation for the *newrelic.yml* file on the New Relic website, and inserted all of the non-deprecated configuration options, alphabetized, with documentation, into the config file template.

`newrelic install` will now install a *Devise* style, well-documentated configuration file.

# Related Github Issue
#684 

# Testing
We confirmed that the generated config file has all of the same uncommented sections that the old, much shorter version has. So there should be no operational effect to using this longer version.

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

**Comment by wyhaines** [comment]
That spec failure looks like it is unrelated to the PR. Is that correct?

**Comment by tannalynn** [comment]
> That spec failure looks like it is unrelated to the PR. Is that correct?

Yes, that's correct. The infinite tracing tests are a bit flaky due to some multithreading issues with our fake trace observer we use for those tests. It's been improved over time, but sometimes it still causes some issues. I've started a rerun on those jobs, but it looks like github actions is having some issues setting up our test environments right now. I'll try again in a bit.

**Review by tannalynn — CHANGES_REQUESTED** [review]
Thanks so much for working on this one!

The (Dynamic) thing in the docs on some config options default value indicates that the agent decides what the value will be once it starts up (this is for various reasons, depending on the specific config option). For all config options that have that, it might be good to change the example value to something the user would be able to use if uncommented maybe? I mentioned the instrumentation ones in another comment, which is most of them, but the others have various possibilities. Some are simply booleans, so that should be easy to change, some have the options listed in the description (like `slow_sql.record_sql` for example). If you aren't sure about a specific one, just let me know and I'd be happy to help figure out what a good example value to set would be.

File: newrelic.yml [hunk 1]
```diff
@@ -13,15 +13,600 @@ common: &default_settings
...
+# tail-based sampling by sending all recorded spans to a Trace Observer for
+# further sampling decisions, irrespective of any usual agent sampling decision.
+# infinite_tracing.trace_observer.host: ""
+
+# Configures the TCP/IP port for the Trace Observer Host
+# infinite_tracing.trace_observer.port: 443
+
+# Controls auto-instrumentation of bunny at start up.
+# May be one of [auto|prepend|chain|disabled].
+# instrumentation.bunny: (Dynamic)
```
[the remark this answers is not in the document]
Comment (creaturenex) [reply newrelic.yml:309]: Hello Tanna,

Not a problem, you make a good point.

File: newrelic.yml [hunk 2]
```diff
@@ -13,15 +13,600 @@ common: &default_settings
...
+# disable_vm_sampler: false
+
+# Distributed tracing lets you see the path that a request takes through your
+# distributed system. Enabling distributed tracing changes the behavior of some
+# New Relic features, so carefully consult the transition guide before you
+# enable this feature.
+# distributed_tracing.enabled: false
+
+# The Entity GUID for the entity that is running this agent.
+# entity_guid: nil
```
[the remark this answers is not in the document]
Comment (creaturenex) [reply newrelic.yml:213]: ill update that

**Comment by creaturenex** [comment]
@wyhaines @tannalynn 
I have updated all references to Dynamic. Please review them to see if my default values makes sense for each setting. I am happy to explain my thought process for any of them. Let me know what you think.

**Review by tannalynn — APPROVED** [review]
Looks great! Thank you for getting those Dynamic values updated!

**Pull request merged**